roadmap/implementers-guide/src/types/overseer-protocol.md

```diff
@@ -236,6 +236,30 @@ enum ChainApiMessage {
 		/// The response channel.
 		response_channel: ResponseChannel<Result<Vec<Hash>, Error>>,
 	}
+	/// Blacklist a set of block hashes to be considered poisened.
+	/// Assumes all hashes exist only once in the set.
+	Blacklist{
```

Comment:
IMO it would make sense to create a new subsystem that would have a dedicated db like av-store.

Reply:
There is already a runtime API that does this :)

Reply:
This is clarified in DM, it will be based on https://github.com/paritytech/substrate/blob/0840c58849bc84e7fd72627745bda243741adcd8/client/service/src/client/client.rs#L1076, but needs some changes to make it persistent.